csrc/id_model/id_model.h

```diff
@@ -116,7 +116,7 @@ class IdModel : public PolymorphicBase {
       const std::vector<Expr*>& exprs,
       const std::vector<TensorView*>& additional_tvs = {},
       bool build_graphs = false,
-      bool allow_self_mapping = false,
+      bool allow_self_mapping = true,
```

Comment:
Changed to not check self mapping by default since that restriction only matters for indexing.